base/sparse/cholmod.jl

```diff
@@ -860,7 +860,18 @@ convert(::Type{Dense}, A::Sparse) = sparse_to_dense(A)
 function (::Type{Sparse}){Tv<:VTypes}(m::Integer, n::Integer,
         colptr::Vector{SuiteSparse_long}, rowval::Vector{SuiteSparse_long},
         nzval::Vector{Tv}, stype)
-    # check if columns are sorted
+    # checks
+    ## length of input
+    if length(nzval) != length(rowval)
+        throw(ArgumentError("nzval and rowval must have same length"))
+    end
+    if length(colptr) != n + 1
+        throw(ArgumentError("length of colptr must be n + 1 = $(n + 1) but was $(length(colptr))"))
+    end
+    if colptr[end] != length(rowval)
```

Comment:
I can see why you don't want this so I'll change it. However, I think I might have assumed this all over the CHOLMOD wrappers so there might be other places where the number of non-zeros is queried with `length(nzval)` which is then wrong.